File: 11-container-with-most-water/container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int l=0, r=height.size()-1;
        int res = 0;
        while(l < r) {
            cout << l << " " << r << " " << res <<  endl;
            int cur = min(height[l], height[r]) * (r - l);
            if(height[l] < height[r]) {
                l++;
            }
            else{
                r--;
            }
            res = max(res, cur);
        }
        return res;
    }
};

File: 11-container-with-most-water/container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int l = 0;
        int r = height.length - 1;
        int res = 0;
        while (l < r) {
            int cur = (r - l) * Math.min(height[r], height[l]);
            if (height[l] > height[r]) {
                r -= 1;
            } else {
                l += 1;
            }
            res = Math.max(cur, res);
        }
        return res;
    }
}